src/h5p-vocabulary-drill.ts

```diff
@@ -7,21 +7,164 @@ import { H5P, H5PContentType, registerContentType } from "h5p-utils";
+    );
+    const toolbar = VocabularyDrill.createToolbar(title, settings);
+
+    containerElement.appendChild(settings);
+    containerElement.appendChild(toolbar);
+    containerElement.appendChild(wrapper);
+    containerElement.classList.add("h5p-vocabulary-drill");
+
+    VocabularyDrill.addRunnable(wrapper, contentId, params);
+  }
+
+  private static createToolbar(
+    title: string,
+    settingsDiv: HTMLDivElement,
+  ): HTMLDivElement {
+    const nodeTitle = document.createTextNode(title);
+    const titleElement = document.createElement("p");
+    titleElement.appendChild(nodeTitle);
+
+    const toolbar = document.createElement("div");
+    toolbar.classList.add("h5p-vocabulary-drill-toolbar");
+    toolbar.appendChild(titleElement);
+
+    const button = document.createElement("button");
+    button.addEventListener("click", () =>
+      settingsDiv.classList.toggle("visible"),
+    );
+
+    toolbar.appendChild(button);
+    return toolbar;
+  }
+
+  private static createSettings(
+    handleAnswerModeChange: () => void,
+    handleLanguageModeChange: () => void,
+  ): HTMLDivElement {
+    const settings = document.createElement("div");
+    settings.classList.add("h5p-vocabulary-drill-settings");
+
+    const buttonClose = document.createElement("button");
+    buttonClose.addEventListener("click", () =>
+      settings.classList.toggle("visible"),
+    );
+
+    const top = document.createElement("div");
+    top.classList.add("h5p-vocabulary-drill-settings-top");
+    top.appendChild(buttonClose);
+
+    const container = document.createElement("div");
+    container.classList.add("h5p-vocabulary-drill-settings-container");
+
+    // TODO: translate
+    const nodeAnswerMode = document.createTextNode("Change answer mode");
+    const buttonAnswerModeLabel = document.createElement("p");
+    buttonAnswerModeLabel.appendChild(nodeAnswerMode);
+
+    const buttonAnswerMode = document.createElement("button");
+    buttonAnswerMode.addEventListener("click", handleAnswerModeChange);
+    buttonAnswerMode.addEventListener("click", () =>
+      settings.classList.toggle("visible"),
+    );
+    buttonAnswerMode.appendChild(buttonAnswerModeLabel);
+
+    // TODO: translate
+    const nodeLanguageMode = document.createTextNode("Change language");
+    const buttonLanguageModeLabel = document.createElement("p");
+    buttonLanguageModeLabel.appendChild(nodeLanguageMode);
+
+    const buttonLanguageMode = document.createElement("button");
+    buttonLanguageMode.addEventListener("click", handleLanguageModeChange);
+    buttonLanguageMode.addEventListener("click", () =>
+      settings.classList.toggle("visible"),
+    );
+    buttonLanguageMode.appendChild(buttonLanguageModeLabel);
+
+    container.appendChild(buttonAnswerMode);
+    container.appendChild(buttonLanguageMode);
+
+    settings.appendChild(top);
+    settings.appendChild(container);
+
+    return settings;
+  }
+
+  private async handleAnswerModeChange(): Promise<void> {
```

Comment:
Why async?

Reply:
Good point, don't need it so I'll change it